lightkurve/lightcurvecollection.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import division, print_function
+
+import copy
+from tqdm import tqdm
+import os
+import datetime
+import logging
+from collections import Sequence
+
+import oktopus
+import numpy as np
+from scipy import signal
+from scipy.optimize import minimize
+from matplotlib import pyplot as plt
+import matplotlib as mpl
+
+from astropy.stats import sigma_clip
+from astropy.table import Table
+from astropy.io import fits
+from astropy.time import Time
+
+
+__all__ = ['LightCurve', 'KeplerLightCurve', 'TessLightCurve',
+           'iterative_box_period_search']
+
+log = logging.getLogger(__name__)
+
+class LightCurveCollection(Sequence):
+    """
+    Collects multiple LightCurve objects together with helpful functions.
+    """
+
+    def __init__(self, lcs):
+        lightcurves = np.asarray(lcs)
+        self.data = {}
+        for lc in lightcurves:
+            if lc.keplerid:
+                self.data[lc.keplerid] = lc
+
+    def __len__(self):
+        return len(self.data)
+
+    def _ids(self):
+        '''
+        Returns the kepler_ids of all the lightcurves as a dict_keys obj.
+        '''
+        return self.data.keys()
+
+    def __getitem__(self, kep_id):
+        '''
+        Returns the lightcurve associated with the kepler_id. 
+        '''
+        try: 
+            return self.data[kep_id]
+        except:
+            raise ValueError('No LightCurve for ' + kep_id)
+
+    def append(self, lc):
+        try:
+            self.data[lc.keplerid] = lc
+        except:
+            log.warning("Input is not a lightcurve")
```

Comment:
This should raise an exception not a warning.

Reply:
Done